src/types/list.rs

```diff
@@ -18,30 +20,79 @@ pub struct PyList(PyAny);
 pyobject_native_type_core!(PyList, ffi::PyList_Type, #checkfunction=ffi::PyList_Check);
 
 #[inline]
-unsafe fn new_from_iter<T>(
-    elements: impl ExactSizeIterator<Item = T>,
+#[track_caller]
+fn new_from_iter<T>(
+    py: Python,
+    mut elements: impl ExactSizeIterator<Item = T>,
     convert: impl Fn(T) -> PyObject,
 ) -> *mut ffi::PyObject {
```

Comment:
This function could return `Py<PyList>` to make clear that the pointer is owned. What do you think?

Reply:
Agreed, I'd support that!